packages/react-components/src/components/table-renderer-components.tsx

```diff
@@ -139,45 +144,84 @@ export class SearchFilterRenderer extends React.Component<SearchFilterRendererPr
                             autoFocus={true}
                             onKeyDown={this.onKeyDownEvent}
                             onInput={this.onInputBoxChanged}
-                            style={{ width: '50%', margin: '10px' }}
+                            style={{
+                                width: '50%',
+                                margin: '10px',
+                                color: this.state.isSearchSuccessful ? '' : 'red'
+                            }}
                             defaultValue={this.props.filterModel.get(this.props.colName) ?? ''}
+                            title="Enter a regular expression, then press Enter"
                         />
+                        {this.state.isSearching && (
+                            <FontAwesomeIcon
+                                spin
+                                icon={faSpinner}
+                                style={{ marginRight: '10px', marginTop: '20px' }}
+                                title="Searching..."
+                            />
+                        )}
                         <FontAwesomeIcon
                             className="hoverClass"
                             icon={faTimes}
                             style={{ marginTop: '20px' }}
                             onClick={this.onCloseClickHandler}
+                            title="Clear search"
                         />
                         <FontAwesomeIcon
                             className="hoverClass"
                             icon={faAngleDown}
                             style={{ marginLeft: '10px', marginTop: '20px' }}
                             onClick={this.onDownClickHandler}
+                            title="Find next"
                         />
                         <FontAwesomeIcon
                             className="hoverClass"
                             icon={faAngleUp}
                             style={{ marginLeft: '10px', marginTop: '20px' }}
                             onClick={this.onUpClickHandler}
+                            title="Find previous"
                         />
                     </div>
                 )}
             </div>
         );
     }
 
+    private async onFilterChange(colName: string, inputVal: string) {
+        this.setState({ isSearchSuccessful: true, isSearching: false });
+        this.props.onFilterChange(colName, inputVal);
+    }
+
+    private runSearch(callback: () => Promise<boolean> | undefined): void {
+        this.setState({ isSearching: true, isSearchSuccessful: true }, () => {
+            this.forceUpdate(() => {
+                const result = callback();
+                if (result === undefined) {
+                    return;
+                }
+                result.then(isFound => {
```

Comment:
Should we handle also the reject case, i.e. reset the flags? Rejecting is not implemented, but to be future proof. WDYT?

Reply:
I could tack on: 

```ts
                result.catch(() => {
                    this.setState({
                        isSearching: false,
                        isSearchSuccessful: false,
                    });
                });
```

Reply:
Should I worry about error messages / the like? I think resetting the flags should be sufficient.

Reply:
That's what I had in mind. Reset the flags to avoid being in an incorrect state. No error message is ok.

Reply:
Just added it in! One of my seniors over here wanted me to leave an error message in since it might be handy down the road. 

Thanks again for your help :)

Reply:
I agree with your colleague that logging errors is important and we should do it. However, there is no proper logging infrastructure in place at them moment and we avoid using console for that. 

If I approve the PR, there is a good chance that the 2nd reviewer (another committer in project) will ask to remove it. Maybe remove it right away.

Reply:
Should be good to go now.